src/pretix/presale/checkoutflow.py

```diff
@@ -605,10 +630,96 @@ def forms(self):
                 formset.append(formsetentry)
         return formset
 
+    @property
+    def cross_selling_applicable_categories(self):
+        cart = self.positions
+        return [
+            (c, products_qs, discount_info) for (c, products_qs, discount_info) in
+            (
+                (c, *c.cross_sell_visible(cart, self.request.sales_channel.identifier))
+                for c in self.request.event.categories.filter(cross_selling_mode__isnull=False)
+            )
+            if products_qs is not None
+        ]
+
+    @cached_property
+    def cross_selling_is_applicable(self):
+        return any(len(items) > 0 for (category, items) in self.cross_selling_data)
+
+    @cached_property
+    def cross_selling_data(self):
+        class DummyCategory:
+            def __init__(self, rule, subevent=None):
```

Comment:
Why "rule"? Took me a while to understand what is passed in there (type annotation would also help understanding here)

Reply:
that was a leftover from the pretix_cross_selling plugin where I used cross-selling-rules instead of categories